internal/api/v1beta1/app_types.go

```diff
@@ -664,6 +667,34 @@ func (app *App) DoRollback() {
 	app.Spec.Canary.Active = false
 }
 
+// AddLabel adds a label to an app's deployments' processes. It will remove labels with matching keys and targets.
+func (app *App) AddLabel(label map[string]string, target Target) {
+	// clean up labels
+	for i := len(app.Spec.Labels) - 1; i >= 0; i-- {
+		if app.Spec.Labels[i].Target != target {
+			continue
+		}
+		for key := range app.Spec.Labels[i].Apply {
```

Comment:
that's a very tricky function
should it be 
```golang
for key := range labels {
    delete(app.Spec.Labels[i].Apply, key)
}
```
?

Reply:
Good catch. Changed.